### PR TITLE
Clean up test cases in pipelinerun_test.go

### PR DESCRIPTION
**Pull request by ImJasonH**
/kind cleanup

Rather than creating a bunch of fake PipelineRuns that are each individually used by each test case, only provide the single PipelineRun used by each test case. This makes it clearer that each test case is only concerned with a single PipelineRun.

The provided fake Pipelines and Tasks could also be only provided as-needed, but some test cases share Pipelines/Tasks with other test cases, so this is a bit less straightforward.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [y] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [y] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [y] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 1]
```diff
@@ -518,184 +516,163 @@ func TestReconcile_InvalidPipelineRuns(t *testing.T) {
...
 		},
-	}
-
-	for _, tc := range tcs {
+	}} {
 		t.Run(tc.name, func(t *testing.T) {
-
 			d := test.Data{
-				PipelineRuns: prs,
+				PipelineRuns: []*v1beta1.PipelineRun{tc.pipelineRun}, // TODO you changed this
```
Comment (pritidesai) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:675]: 🤔

Comment (ImJasonH) [reply pkg/reconciler/pipelinerun/pipelinerun_test.go]: 😅

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Pull request merged**